test/functional/JIT_Test/src/jit/test/jitserver/JITServerTest.java

```diff
@@ -490,4 +594,78 @@ public void testServerComesUpAfterClientAndGoesDownAgain() throws IOException, I
 		logger.info("Stopping client...");
 		destroyAndCheckProcess(client, clientBuilder);
 	}
+
+	public void testServerSCC() throws IOException, InterruptedException {
+                logger.info("running testServerSCC: INFO and above level logging enabled");
+
+		// Run this tests only for the test variation with AOT Cache option specified
+		if(System.getProperty("CLIENT_PROGRAM").contains(aotCacheOption)) {
+			redirectProcessOutputs(clientBuilder, "testServerSCC.client");
+			redirectProcessOutputs(serverBuilder, "testServerSCC.server");
+
+			updateJITServerPort();
+
+			final Process server = startProcess(serverBuilder, "server");
+
+			Thread.sleep(SERVER_START_WAIT_TIME_MS);
+
+			Process client = startProcess(clientBuilder, "client");
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Check whether the cache test_jitscc exists.");
+			if(checkCacheExists("test_jitscc"))
+				logger.info("Check if AOT methods exists in the cache test_jitscc::" + checkAOTCompileExists("test_jitscc"));
+			else
+				AssertJUnit.fail("The cache test_jitscc is not created.");
+
+			logger.info("Stopping client...");
+			destroyAndCheckProcess(client, clientBuilder);
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Destroy the SCC and then restart the client
+			logger.info("Destroy the cache test_jitscc");
+			destroyCache("test_jitscc");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Check if the SCC exists and then restart the client
+			logger.info("Check whether the cache test_jitscc exists.");
+			if (checkCacheExists("test_jitscc"))
+				AssertJUnit.fail("The cache test_jitscc should have been destroyed.");
+
+			logger.info("Restarting client...");
+			client = startProcess(clientBuilder, "client");
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Stopping client...");
+			destroyAndCheckProcess(client, clientBuilder);
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Destroy the SCC and then restart the client
+			logger.info("Destroy the cache test_jitscc");
+			destroyCache("test_jitscc");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Restarting client...");
+			client = startProcess(clientBuilder, "client");
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			if(!checkLogFiles("testServerSCC.client.jitverboselog", "remote deserialized"))
+				AssertJUnit.fail("There are no deserialized methods at the client.");
+
+			logger.info("Stopping client...");
+			destroyAndCheckProcess(client, clientBuilder);
+
+			logger.info("Stopping server...");
+			destroyAndCheckProcess(server, serverBuilder);
```

Comment:
If the test is successful, don't we have to do some cleanup: destroy any SCC you have created and any vlog files that were created?

Reply:
The verbose logs were getting created in a unique directory by the existing testing code. I just added more data in the existing logging command with 'CompilePerformance' . In my code I just read the files which were getting produced by the existing test code. The verbose and other test logs are not cleared by the existing test code.
The SCC is always deleted by the test code before every test:

```
testJITServer_2_PASSED

"/home/dev/bootjdk8/bin/java" -Xshareclasses:destroyAll; "/home/dev/bootjdk8/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";

Attempting to destroy all caches in cacheDir /root/javasharedresources/

JVMSHRC806I Compressed references persistent shared cache "sharedcc_root" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
JVMSHRC806I Compressed references persistent shared cache "test_jitscc" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
JVMSHRC005I No shared class caches available
cache cleanup done
testJITServer_2 Finish Time: Wed Jan  4 17:50:05 2023 Epoch Time (ms): 1672883405857
```

Reply:
> The SCC is always deleted by the test code before every test:

I was thinking that it's not nice to leave unwanted SCC files on the machines where this test runs. I see some value in leaving the SCC if the test fails (for more diagnostics)

Reply:
Destroyed the SCC after stopping the client.